source/s3_auto_ranged_put.c

```diff
@@ -421,29 +410,21 @@ static void s_s3_meta_request_auto_ranged_put_destroy(struct aws_s3_meta_request
 
     aws_s3_paginated_operation_release(auto_ranged_put->synced_data.list_parts_operation);
 
-    for (size_t etag_index = 0; etag_index < aws_array_list_length(&auto_ranged_put->synced_data.etag_list);
-         ++etag_index) {
-        struct aws_string *etag = NULL;
+    for (size_t part_index = 0; part_index < aws_array_list_length(&auto_ranged_put->synced_data.part_list);
+         ++part_index) {
 
-        aws_array_list_get_at(&auto_ranged_put->synced_data.etag_list, &etag, etag_index);
-        aws_string_destroy(etag);
+        struct aws_s3_put_part_info *part;
+        aws_array_list_get_at(&auto_ranged_put->synced_data.part_list, &part, part_index);
+        if (part) {
```

Comment:
super minor nit: within this pr both if (a) and if (a != NULL) are used. i know we are not super consistent with it in general, but i think we should prefer one style. from my experience c-s3 does != NULL more often

Reply:
good point, I'll go explicit